packages/core/contracts/financial-templates/common/FundingRatePayer.sol

```diff
@@ -72,20 +72,18 @@ abstract contract FundingRatePayer is FeePayer, PerpetualInterface {
         );
         FixedPoint.Unsigned memory collateralPool = _pfc();
 
-        // If contract has no PfC, then cannot pay any fees.
-        require(!collateralPool.isEqual(0), "PfC is 0");
-        // The fee must be non-zero.
-        require(!amount.isEqual(0), "Funding fee is 0");
-        // The fee must be < available collateral.
-        require(collateralPool.isGreaterThan(amount), "Funding fee is more than PfC");
+        // If amount to pay or collateral pool is 0, then return early.
+        if (!amount.isEqual(0) && !collateralPool.isEqual(0)) {
```

Comment:
small nit here but why not return early like the comment suggests?

Reply:
Ah confusing comment I guess, if the condition is met (i.e. amount to pay or collateral pool is 0) it skips to the end of the if statement and basically does nothing.

Reply:
Thoughts on flipping it to work like this:
```js
if (amount.isEqual(0) || collateralPool.isEqual(0)) {
  return;
}

FixedPoint.Unsigned memory amountToPay = (amount.isGreaterThan(collateralPool) ? collateralPool : amount);
...
```

IMO, that makes the non-payment case a little more obvious and it makes the conditional in the if statement a little simpler.

Reply:
sure

Reply:
Done